Automattic/newspack-plugin
# Managed Newspack Wizard

### All Submissions:






### Changes proposed in this Pull Request:



_Note: This branch adds a new Composer dependency (`stripe/stripe-php`), so be sure to run `composer update` before testing._

Adds a wizard for managed Newspack customers to input or change billing information. This wizard will not be visible unless three `wp-config` constants are set:

```
define( 'NEWSPACK_STRIPE_SECRET_KEY', 'sk_...' );
define( 'NEWSPACK_STRIPE_PUBLISHABLE_KEY', 'pk_...' );
define( 'NEWSPACK_STRIPE_PLAN', 'plan_...' );
```

`NEWSPACK_STRIPE_SECRET_KEY`: Stripe secret key.
`NEWSPACK_STRIPE_PUBLISHABLE_KEY`: Stripe publishable key.
`NEWSPACK_STRIPE_PLAN`: The id of the plan to which the current publication should be subscribed.

If all three are set `Managed Newspack` will appear at the bottom of the list of wizards in the left nav. In the Wizard, clicking Subscribe redirects to a Stripe Checkout page where credit card info can be input. When the form is submitted, the page redirects back to the wizard and the Stripe customer ID and subscription ID will be stored in the WP database. If a subscription/customer has been created the Wizard shows an `ActionCard` with some information about the subscription. Clicking `Update` redirects to Stripe Checkout where a new payment option can be input, replacing the existing one.

<img width="1726" alt="Screen Shot 2020-02-25 at 12 22 14 AM" src="https://user-images.githubusercontent.com/1477002/75217482-260e3480-5765-11ea-8990-42d7a2f7ba27.png">

<img width="1726" alt="Screen Shot 2020-02-25 at 12 22 22 AM" src="https://user-images.githubusercontent.com/1477002/75217501-33c3ba00-5765-11ea-87aa-720a06debf6d.png">

<img width="1726" alt="Screen Shot 2020-02-25 at 12 22 49 AM" src="https://user-images.githubusercontent.com/1477002/75217512-3a523180-5765-11ea-9b0d-87d43b47e785.png">

This Wizard could be expanded to include support features, cc: @adekbadek.

### How to test the changes in this Pull Request:

1. Log in to Stripe.
2. Switch `Viewing Test Data` on.
3. Navigate to Developers->API Keys and copy the Publishable and Secret Keys.
4. Navigate to Billing->Products, edit `Managed Newspack` and copy the id for one of the Pricing Plans.
5. Add these three values to test site's `wp-config.php`, as described above.
6. In WP, click the Newspack icon in the left nav bar.
7. Verify `Managed Newspack` is shown at the bottom of the list of wizards.
8. Click Managed Newspack. 
9. Click Subscribe. 
10. Input a test Stripe card (https://stripe.com/docs/testing), and fill in all other information (can be fake).
11. After submitting, verify you're redirected to the Wizard, and the details of the subscription are shown. 
12. In the Stripe dashboard, verify your customer and subscription have been created.
13. In the Newpsack Wizard, click `Update`.
14. Input a different  credit card.
15. In Stripe dashboard verify the second payment method has been added, and is the new default.
16. In Stripe Dashboard, delete the customer you created.
17. In Newspack Wizard, verify return to original pre-subscription state.

### Other information:
